python/tchannel/tornado/message_factory.py

```diff
@@ -315,10 +329,40 @@ def fragment(message):
                 payload_space = (common.MAX_PAYLOAD_SIZE -
                                  rw.length_no_args(message))
                 fragment_msg = message.fragment(payload_space)
+                self.generate_checksum(message, message_id)
                 yield message
         else:
             yield message
 
+    def generate_checksum(self, message, message_id):
+        if message.message_type not in CHECKSUM_MSG_TYPES:
+            return
+        generate_checksum(
+            message,
+            self.out_checksum.get(message_id, 0),
+        )
+
+        self.out_checksum[message_id] = message.checksum[1]
+        if message.flags == FlagsType.none:
+            self.out_checksum.pop(message_id)
+
+    def verify_message(self, message, message_id):
+        """Verify the checksum of the message."""
+        if verify_checksum(
+                message,
+                self.in_checksum.get(message_id, 0),
+        ):
+            self.in_checksum[message_id] = message.checksum[1]
+
+            if message.flags == FlagsType.none:
+                self.in_checksum.pop(message_id)
+        else:
+            try:
+                self.in_checksum.pop(message_id)
+            except KeyError:
+                pass
```

Comment:
Please don't use exceptions for control flow. There's a significant performance hit to that. Check if `message_id in self.in_checksum` manually.